opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -11,3 +11,513 @@
+            measurements: The collection of measurements to record. options:
+            An optional :class:`.RecordOptions` used to populate details during
+            recording.
+
+        Returns: None
+        """
+
+
+_METER: typing.Optional[Meter] = None
+_METER_FACTORY: typing.Optional[
+    typing.Callable[[typing.Type[Meter]], typing.Optional[Meter]]] = None
+
+
+def meter() -> Meter:
+    """Gets the current global :class:`.Meter` object.
+
+    If there isn't one set yet, a default will be loaded.
+    """
+    global _METER, _METER_FACTORY  # pylint:disable=global-statement
+
+    if _METER is None:
+        # pylint:disable=protected-access
+        _METER = loader._load_impl(Meter, _METER_FACTORY)
+        del _METER_FACTORY
+
+    return _METER
+
+
+def set_preferred_meter_implementation(
+        factory: typing.Callable[
+            [typing.Type[Meter]], typing.Optional[Meter]]
+        ) -> None:
+    """Set the factory to be used to create the :class:`.Meter`.
+
+    See :mod:`opentelemetry.loader` for details.
+
+    This function may not be called after a meter is already loaded.
+
+    Args:
+        factory: Callback that should create a new :class:`.Meter` instance.
+    """
+    global _METER_FACTORY  # pylint:disable=global-statement
+
+    if _METER:
+        raise RuntimeError("Meter already loaded.")
+
+    _METER_FACTORY = factory
+
+
+class LabelKey:
+
+    def __init__(self,
+                 key: str,
+                 description: str) -> None:
+        self.key = key
+        self.description = description
+
+
+class LabelValue:
+
+    def __init__(self,
+                 value: str) -> None:
+        self.value = value
+
+
+class MetricOptions:
+    def __init__(self,
+                 description: str,
+                 unit: str,
+                 label_keys: typing.List['LabelKey'],
+                 constant_labels: typing.Dict['LabelKey', 'LabelValue'],
+                 component: str,
+                 resource: 'Resource') -> None:
+        """Optional info used when creating a :class:`.Metric`.
+
+        Args:
+            description: Human readable description of the metric.
+            unit: Unit of the metric values.
+            label_keys: list of keys for the labels with dynamic values.
+            Order of the list is important as the same order MUST be used
+            on recording when suppling values for these labels
+            constant_labels: A map of constant labels that will be used for
+            all of the TimeSeries created from the Metric.
+            component: The name of the component that reports this metric.
+            Resource: Sets the :class:`.Resource` associated with this
+            Metric.
+        """
+        self.description = description
+        self.unit = unit
+        self.label_keys = label_keys
+        self.constant_labels = constant_labels
+        self.component = component
+        self.resource = resource
+
+
+class MeasureType:
```

Comment:
Hopefully this is one of those ugly things we only need in javascript...

Reply:
Do you think we should have some other way of representing the type?

Reply:
I may be missing something here, but ideally we represent differently typed metrics with different classes.